ts/components/leftpane/ActionsPanel.tsx

```diff
@@ -149,14 +147,25 @@ const cleanUpMediasInterval = DURATION.MINUTES * 60;
 const fetchReleaseFromFileServerInterval = 1000 * 60; // try to fetch the latest release from the fileserver every minute
 
 const setupTheme = async () => {
+  const shouldFollowSystemTheme = window.getSettingValue(SettingsKey.hasFollowSystemThemeEnabled);
   const theme = window.Events.getThemeSetting();
-  // We don't want to reset the primary color on startup
-  await switchThemeTo({
+  const themeConfig = {
     theme,
     mainWindow: true,
     usePrimaryColor: true,
     dispatch: window?.inboxStore?.dispatch || undefined,
-  });
+  };
+
+  if (shouldFollowSystemTheme) {
+    const themeIsCongruent = await checkThemeCongruency();
+    // Only set theme if it matches with native theme, otherwise handled by checkThemeCongruency()
+    if (!themeIsCongruent) {
+      await switchThemeTo(themeConfig);
+    }
+    return;
+  }
+
+  await switchThemeTo(themeConfig);
```

Comment:
not sure which approach you want to take

Reply:
So in this function setupTheme() the theme must be set, and the ensureThemeConsistency() function does not guarantee  that it will set the theme, it only sets the theme if there is a mismatch between system and currently set theme. So instead the function returns whether it set the theme or not. Based on this i can set the theme if it wasn't set. Maybe i can split this ensureThemeConsistency() into two functions if you think that's required